core/src/sumeragi/mod.rs

```diff
@@ -348,24 +325,13 @@ impl Sumeragi {
     /// Update the sumeragi internal online peers list.
     #[allow(clippy::expect_used)]
     pub fn update_online_peers(&self, online_peers: Vec<PeerId>) {
-        *self
-            .internal
-            .current_online_peers
-            .lock()
-            .expect("Failed to lock on update online peers.") = online_peers;
+        *self.internal.current_online_peers.lock_mut() = online_peers;
     }
 
     /// Deposit a sumeragi network message.
     #[allow(clippy::expect_used)]
     pub fn incoming_message(&self, msg: MessagePacket) {
-        if self
-            .internal
-            .message_sender
-            .lock()
-            .expect("Lock on sender")
-            .try_send(msg)
-            .is_err()
-        {
+        if self.internal.message_sender.lock().try_send(msg).is_err() {
```

Comment:
The `if` on an error variant discards the error message. If you do an `if let Err(err)` you can also print the error message in `debug!`, so that we have a fuller picture.